gr-digital/include/gnuradio/digital/adaptive_algorithm.h

```diff
@@ -0,0 +1,90 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifndef INCLUDED_DIGITAL_ADAPTIVE_ALGORITHM_H
+#define INCLUDED_DIGITAL_ADAPTIVE_ALGORITHM_H
+
+#include <gnuradio/digital/api.h>
+#include <gnuradio/digital/constellation.h>
+#include <gnuradio/math.h>
+#include <math.h>
+#include <algorithm>
+#include <vector>
+
+namespace gr {
+namespace digital {
+enum class adaptive_algorithm_t { LMS, NLMS, CMA };
+
+class adaptive_algorithm;
+typedef std::shared_ptr<adaptive_algorithm> adaptive_algorithm_sptr;
```

Comment:
For this class I copied the way it was done in constellation.h where the base class had a unique *_sptr perhaps because it wouldn't be instantiated directly?

Reply:
I don't know. Seems like a style issue more than anything, and not one I can answer.